docs/docs/guides/integrations/mcp.md

```diff
@@ -0,0 +1,236 @@
+```
+
+This will start the client, which will connect to and interact with the server. The client provides a command-line interface with the following options:
+
+- `tools` - List all available tools
+- `resources` - List all available resources
+- `prompts` - List all available prompts
+- `add <a> <b>` - Add two numbers
+- `bmi <weight_kg> <height_m>` - Calculate BMI
+- `weather <city>` - Get weather for a city
+- `greeting <name>` - Get personalized greeting
+- `user <id>` - Get user profile
+- `config` - Get application configuration
+- `code-review <code>` - Get code review prompt
+- `debug <error>` - Get debug error prompt
+- `demo` - Run demos for all features
+- `q` - Exit the session
+
+If you type `demo`, the client will run through all the features. Typing `q` will close the process. If you want to play with the available features, try individual commands listed above.
+
+You will find a trace timeline as shown below:
+
+[![mcp_trace_timeline.png](imgs/mcp/mcp_trace_timeline.png)](https://wandb.ai/ayut/mcp_example/weave/traces?filter=%7B%22opVersionRefs%22%3A%5B%22weave%3A%2F%2F%2Fayut%2Fmcp_example%2Fop%2Frun_client%3A*%22%5D%7D&peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bbe-cc5e-7012-b45f-bf10617d8c1e%3FhideTraceTree%3D0)
+
+By default, you will just see the `run_client` traces. Click on the Ops selection box and select "All Calls"
+
+[!mcp_all_calls.png](imgs/mcp/mcp_all_calls.png)]
+
+Doing so will show you `FastMCP` methods (tools, resources, prompts) traced by the integration. You can see the arguments given to the tools and returned values.
+
+[!mcp_fastmcp.png](imgs/mcp/mcp_fastmcp.png)](https://wandb.ai/ayut/mcp_example/weave/traces?peekPath=%2Fayut%2Fmcp_example%2Fcalls%2F01966bc2-aca1-7021-a626-aecfe677b1b4%3FhideTraceTree%3D0)
+
+### Understanding the Example
+
+#### Server-Side Code
+
+The example server (`example_server.py`) defines:
+
+1. **Tools**: Functions like `add()`, `calculate_bmi()`, and `fetch_weather()`
+2. **Resources**: Data endpoints like `greeting://{name}`, `config://app`, and `users://{user_id}/profile`
+3. **Prompts**: Templates like `review_code()` and `debug_error()`
+
+All of these components are automatically traced by Weave, allowing you to monitor their usage, inputs, and outputs.
+
+#### Client-Side Code
+
+The example client (`example_client.py`) demonstrates:
+
+1. How to connect to an MCP server
+2. How to discover available tools, resources, and prompts
+3. How to call tools with parameters
+4. How to access resources with URIs
+5. How to generate prompts with arguments
+
+All client-side operations are also traced by Weave, creating a complete picture of the client-server interaction.
+
+## Using the Integration
+
+### Client-Side Integration
+
+To enable tracing on an MCP client:
+
+```python
+import weave
+from mcp import ClientSession, StdioServerParameters
+from mcp.client.stdio import stdio_client
+
+# Initialize Weave
+weave_client = weave.init("my-project")
```

Comment:
use code highlighting here to highlight the weave.init call and say that its all you need

add `# highlight-next-line` to the line above what you want to highlight